synapse/lib/cell.py

```diff
@@ -67,6 +67,25 @@ def getCellIden(self):
     def getCellUser(self):
         return self.user.pack()
 
+    def setCellUser(self, iden):
+        '''
+        Switch to another user (admin only).
+
+        This API allows remote admin/service accounts
+        to impersonate a user.  Used mostly by services
+        that manage their own authentication/sessions.
+        '''
+        if not self.user.admin:
```

Comment:
If a user switches to a lower privileged user, they will be unable to switch back out of it.

Reply:
Exactly correct.  Completely intended.